torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -283,6 +285,34 @@ c10::IValue PyRRef::toIValue() const {
   return IValue(rrefPtr);
 }
 
+void PyRRef::backward(int64_t dist_autograd_ctx_id, bool retain_graph) {
+  if (rref_->isOwner()) {
+    auto value =
+        c10::static_intrusive_pointer_cast<const OwnerRRef>(rref_)->getValue();
+    TORCH_CHECK(
+        value.isTensor(), "RRef should contain a tensor for .backward()");
+    auto root = value.toTensor();
+    TORCH_CHECK(root.requires_grad(), "requires_grad not set on RRef's value");
+    TORCH_CHECK(
+        root.numel() == 1,
+        "RRef does not contain a scalar, all roots need to be scalar");
+    TORCH_CHECK(
+        root.grad_fn(), "RRef does not have a valid gradient function.");
```

Comment:
Curious, do we need these three checks, or did the autograd engine already do that?

Reply:
These were unnecessary, I called the CPP APIs instead and these checks were in that API.